x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/string/LocateTests.java

```diff
@@ -131,6 +132,45 @@ public void testExactString() {
         assertThat(process("界世", "界世", 0), equalTo(1));
     }
 
+    public void testSupplementaryCharacter() {
```

Comment:
I put all these test in one place, because supplementary chars hurt ones head - it's easier to reason about them when they're all together.